src/OutputManager.php

```diff
@@ -132,17 +134,16 @@ private function saveAutoloader($name, array $classes)
         // First we generate a string containing the known classes and the paths they map to. One line for each string.
         $autoloadedClasses = array();
         foreach ($classes as $class) {
-            $className = $this->config->get('namespaceName') . '\\' . $class->getIdentifier();
+            $className = $this->config->get('namespaceName') . '\\' . $class->getName();
             $className = ltrim($className, '\\');
-            $autoloadedClasses[] = "'" . $className . "' => __DIR__ .'/" . $class->getIdentifier() . ".php'";
+            $autoloadedClasses[] = "'" . $className . "' => __DIR__ .'/" . $class->getName() . ".php'";
         }
         $autoloadedClasses = implode(',' . PHP_EOL . str_repeat(' ', 8), $autoloadedClasses);
 
         // Assemble the source of the autoloader function containing the classes and the check to include.
-        // Our custom code generation library does not support generating code outside of functions and we need to
-        // register the autoloader in the global scope. Consequently we manually insert a } to end the autoloader
-        // function, register it and finish with a {. This means our generated code ends with a no-op {} statement.
         $autoloaderSource = <<<EOF
+function $autoloaderName(\$class)
```

Comment:
Yes, it is possible, but is it good? It will look like this:

```php
$autoloader = (new MethodGenerator())
    ->setName($autoloaderName)
    ->setParameter(
        (new ParameterGenerator())
            ->setName('class')
    )
    ->setBody('$classes = array(' .
        '    ' . $autoloadedClasses .
        ');' .
        'if (!empty($classes[$class])) {' .
        '    include $classes[$class];' .
        '};'
    );

$file = (new FileGenerator())
    ->setFilename($this->dir . DIRECTORY_SEPARATOR . 'autoload.php')
    ->setBody(
        $autoloader->generate() . PHP_EOL .
        'spl_autoload_register(\'' . $autoloaderName . '\');'
    )
    ->write();
```

Code is longer and less clear. ```MethodGenerator``` is meant to be used in ```ClassGenerator```, and can't be used in ```FileGenerator``` directly, so any way of assembling whole file will concatenate strings.